### PR TITLE
Add TLS extended_master_secret event

### PR DESCRIPTION
**Pull request by ueno**
As the use of EMS is now mandatory in FIPS 140-3 when TLS 1.2 is negotiated, this would be useful for auditing.

### CODE REVIEW

File: docs/logging-format.md [hunk 1]
```diff
@@ -220,13 +220,14 @@ and TLS probe points.
...
+|------------------------------------|----------------|--------------------------------------------------------------------------------------------------|
+| `tls::protocol_version`            | uint16         | Negotiated TLS version                                                                           |
+| `tls::ciphersuite`                 | uint16         | Negotiated ciphersuite (as in IANA [registry][iana-tls-ciphersuites])                            |
+| `tls::signature_algorithm`         | uint16         | Signature algorithm used in the handshake (as in IANA [registry][iana-tls-signature-algorithms]) |
+| `tls::key_exchange_algorithm`      | uint16         | Key exchange mode: ECDHE(0), DHE(1), PSK(2), ECDHE-PSK(3), DHE-PSK(4)                            |
+| `tls::group`                       | uint16         | Groups used in the handshake (as in IANA [registry][iana-tls-supported-groups])                  |
+| `tls::ext::extended_master_secret` | word (ignored) | Present when extended_master_secret extension is negotiated                                      |
```
Comment (ep69) [inline docs/logging-format.md:230]: It is not clear to me what `word` means in this context. Also, what `(ignored)` means here?

File: crypto-auditing/src/bpf/audit.h [hunk 2]
```diff
@@ -12,7 +12,7 @@ typedef enum
     AUDIT_DATA_BLOB = 2,
   } audit_data_type_t;
 
-#define KEY_SIZE 32
+#define KEY_SIZE 64
```
Comment (ep69) [inline crypto-auditing/src/bpf/audit.h:15]: How is this change related to EMS event?

**Review by ep69 — APPROVED** [review]
Asked couple of questions, but not blocking the merge if it is fine.

File: docs/logging-format.md (hunk 1 above)
Comment (ueno) [reply docs/logging-format.md:230]: `word` means machine word on the host architecture (usually `long`). Since only the presence of this event matters, the value itself is ignored.

File: crypto-auditing/src/bpf/audit.h (hunk 2 above)
Comment (ueno) [reply crypto-auditing/src/bpf/audit.h:15]: The reason behind this is that the key name (`tls::ext::extended_master_secret`) doesn't fit in 32 bytes. Maybe we could use more generic way to represent extension advertisement / acceptance, something like:
`tls::ext::advertised` with extension ID as the value. Let me rewrite in that way.

Comment (ueno) [reply crypto-auditing/src/bpf/audit.h:15]: In the end I gave up on this idea, as the semantics of extension negotiation is defined per extension. So I simply bumped the key length to 64.
